test/core/client_channel/resolvers/binder_resolver_test.cc

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string.h>
+
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
+#include <grpc/support/string_util.h>
+
+#include "src/core/ext/filters/client_channel/resolver_registry.h"
+#include "src/core/lib/channel/channel_args.h"
+#include "test/core/util/test_config.h"
+
+namespace {
+
+class ResultHandler : public grpc_core::Resolver::ResultHandler {
+ public:
+  void ReturnResult(grpc_core::Resolver::Result /*result*/) override {}
+
+  void ReturnError(grpc_error_handle error) override {
+    GRPC_ERROR_UNREF(error);
+  }
+};
+
+void test_succeeds(grpc_core::ResolverFactory* factory, const char* string) {
```

Comment:
Can you please write this as a gunit test instead of adding more C-style tests?  Otherwise, we'll just have to convert it later.

Reply:
Done